### ISSUE TITLE
The chat is responding with external data.

### ISSUE THREAD

**Issue by joaov-lisboas** [Azure-Samples/azure-search-openai-demo#1615]
If the question is unrelated to the scenario I've set up, it correctly states that it doesn't have enough information to respond. However, if it's on the same topic, even if it lacks information, it responds, sometimes citing Wikipedia as a source. I would prefer it to respond only with the data I've provided.

**Comment by pamelafox** [comment]
I have not seen it cite Wikipedia as a source, but I have seen that GPT models may respond off-topic. The best way to mitigate that, in my experience, is to use a more powerful model like GPT-4. I generally recommend that developers use evaluation to determine whether the answer quality is high enough.

I set up this evaluator repo to work well with this app:
https://github.com/Azure-Samples/ai-rag-chat-evaluator

I also added specific tools to measure an app's tendency to answer off-topic:
https://github.com/Azure-Samples/ai-rag-chat-evaluator?tab=readme-ov-file#measuring-apps-ability-to-say-i-dont-know

I do not believe there is much that we can do in this repo, besides default to gpt-4. Each developer must decide for their use case whether it's worth the drawbacks of increased cost and response time, however.

**Comment by joaov-lisboas** [comment]
Thank you for the help, I will work on the evaluator now.
I found a solution, I added the prompt "Respond based exclusively on the data provided below." in the following part
messages = self.get_messages_from_history(
system_prompt=system_message,
model_id=self.chatgpt_model,
history=history,
Model does not handle lengthy system messages well. Moving sources to latest user conversation to solve follow up questions prompt.
user_content=original_user_query + "\n\nRespond based exclusively on the data provided below." + "\n\nSources:\n" + content,
max_tokens=messages_token_limit,
)
In addition, I changed the minimum search score to 0.02 and the minimum reranker score to 1.5.
So far, in my tests, the responses have been satisfactory.
